pkg/csi/manila/controllerserver.go

```diff
@@ -262,8 +262,20 @@ func (cs *controllerServer) CreateSnapshot(ctx context.Context, req *csi.CreateS
 		readyToUse = false
 	case snapshotAvailable:
 		readyToUse = true
+	case snapshotError:
+		manilaErrMsg, err := lastResourceError(snapshot.ID, manilaClient)
+		if err != nil {
+			res.err = fmt.Errorf("snapshot %s of share %s is in error state, error description could not be retrieved: %v", snapshot.ID, req.GetSourceVolumeId(), err)
+			return nil, status.Error(codes.Internal, res.err.Error())
+		}
+
+		// An error occurred, try to roll-back the snapshot
+		tryDeleteSnapshot(snapshot, manilaClient)
```

Comment:
I'm not quite sure about this piece... The `tryDeleteSnapshot` uses exponential backoff -- this means the call to CreateSnapshot might time-out. Wouldn't it be better to run the `tryDeleteSnapshot` in a goroutine so the caller gets an error immediately and the driver attempts to delete the broken snapshot in backgground?

Reply:
We have clarified this with @gman0 on Slack: The timeouts should not cause problems in this case: the CO would retry until some other error is returned or the operation succeeds.